tests/test_clicks.py

```diff
@@ -0,0 +1,115 @@
+from poprox_storage.repositories.clicks import DbClicksRepository
+from poprox_concepts import Account
+from datetime import datetime
+
+import json
+import pytest
+import os
+from sqlalchemy import create_engine, text
+from uuid import uuid4
+
+db_password = os.environ.get("POPROX_DB_PASSWORD", "")
+db_port = os.environ.get("POPROX_DB_PORT", "")
+
+DEFAULT_PG_URL = f"postgresql://postgres:{db_password}@127.0.0.1:{db_port}/poprox"
+
+
+@pytest.fixture(scope="session")
+def pg_url():
+    """
+    Provides base PostgreSQL URL for creating temporary databases.
+    """
+    return os.getenv("CI_POPROX_PG_URL", DEFAULT_PG_URL)
+
+
+def test_get_active_survey(pg_url: str):
+    engine = create_engine(pg_url)
+    with engine.connect() as conn:
+        conn.execute(text("delete from clicks;"))
+        conn.execute(text("delete from newsletters;"))
+        conn.execute(text("delete from articles;"))
+
+        dummy_click_1 = uuid4()
+        dummy_click_2 = uuid4()
+        dummy_user_1 = uuid4()
+        dummy_user_2 = uuid4()
+        dummy_newsletter = uuid4()
+
+        dummy_article_1 = uuid4()
+        dummy_article_2 = uuid4()
+
+        query_0 = text(
+                f'''insert into accounts(account_id, email, status, source) values (:account_id, 'user-1@gmail.com', '', '');'''
+            )
+        query_1 = text(
+                f'''insert into clicks(click_id, account_id, newsletter_id, article_id, created_at)
+                    values (:click_id_1, :account_id, :newsletter_id, :article_id_1, '2024-06-12 09:55:22');'''
+            )
+        query_2 = text(
+                f'''insert into clicks(click_id, account_id, newsletter_id, article_id, created_at)
+                    values (:click_id_2, :account_id, :newsletter_id, :article_id_2,  '2024-07-14 12:55:22');'''
+            )
+        query_3 = text(
+                f'''insert into newsletters(newsletter_id, account_id, content, created_at, html) 
+                values (:newsletter_id, :account_id, '{json.dumps({})}', '{datetime.now()}', '');'''
+            )
+        query_4 = text(
+            f'''insert into articles(article_id, published_at, created_at, title, content, url) 
+                values (:article_id, '{datetime.now()}', '{datetime.now()}', :title, '{json.dumps({})}', :url);'''
+        )
```

Comment:
`query_4` looks like `DbArticleRepository.insert_article()`, which could save you some SQL

Reply:
Hmm, this also does not work as it would throw some weird compilation error to me:
`sqlalchemy.exc.CompileError: Unconsumed column names: preview_image_id, external_id, raw_data, source`

And how I call it is simply created an article object:

`article_id_1 = dbArticleRepository.insert_article(Article(title="Article 1"))`

Not sure what might go wrong, but probably some object-to-sql conversion issue?

Reply:
That is a weird one! That's worth digging into to figure out what's going on; let's do that and sort it out so that you can use that here.